api/src/main/java/org/pmiops/workbench/db/model/User.java

```diff
@@ -130,4 +131,13 @@ public void setFirstSignInTime(Timestamp firstSignInTime) {
   public void setAuthorities(Set<Authority> newAuthorities) {
     this.authorities = newAuthorities;
   }
+
+  @OneToMany(fetch = FetchType.EAGER, mappedBy = "user")
+  public Set<WorkspaceUserRole> getWorkspaceUserRoles() {
+    return workspaces;
+  }
+
+  public void setWorkspaceUserRoles(Set<WorkspaceUserRole> userRoles) {
+    this.workspaces = userRoles;
```

Comment:
Please verify that this deletes the old roles (I think it may not, unless you explicitly remove them from the old `workspaces` first.

Reply:
I may just remove the method body of this, but the actual method needs to be there or JPA complains and it won't run.

Reply:
I would keep the setter, for use when creating a new Workspace (see above.) But we could throw an exception if the current list is already set I think.